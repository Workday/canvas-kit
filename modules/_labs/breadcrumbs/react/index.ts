import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
