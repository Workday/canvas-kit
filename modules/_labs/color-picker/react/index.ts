export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
