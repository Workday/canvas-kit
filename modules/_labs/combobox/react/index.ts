import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
