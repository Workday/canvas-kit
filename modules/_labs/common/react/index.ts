export * from './lib/Flex';
export * from './lib/Box';
