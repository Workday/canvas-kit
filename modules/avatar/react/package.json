{
  "name": "@workday/canvas-kit-react-avatar",
  "version": "3.0.0-alpha.2",
  "description": "A circular user photo (or a default)",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/avatar/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "avatar"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@workday/canvas-kit-react-common": "^3.0.0-alpha.2",
    "@workday/canvas-kit-react-core": "^3.0.0-alpha.2",
    "@workday/canvas-kit-react-icon": "^3.0.0-alpha.2",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12"
  }
}
