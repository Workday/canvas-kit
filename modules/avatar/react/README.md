# Canvas Kit Avatar

A component showing a user's photo with a circular crop.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-avatar
```

## Usage

```tsx
import * as React from 'react';
import {Avatar, AvatarVariant} from '@workday/canvas-kit-react-avatar';

// Basic
<Avatar />

// Using static properties on Avatar
<Avatar
  size={Avatar.Size.xs}
  variant={Avatar.Variant.Dark}
/>

// Using AvatarVariant import directly
<Avatar size={Avatar.Size.xs} variant={AvatarVariant.Dark} />
```

## Static Properties

#### `variant: AvatarVariant`

```tsx
<Avatar variant={Avatar.Variant.Dark} />
```

---

#### `Size: SystemIconCircleSize | number`

```tsx
<Avatar size={Avatar.Size.xs} />

// Custom Size.
// Note: Please use the predefined sizes as it works nicely with the Canvas Design System
<Avatar size={48} />
```

## Component Props

### Required

> None

### Optional

#### `variant: AvatarVariant`

> The variant of the avatar if using a default image.

Default: `AvatarVariant.Light`

| Variant | Description                      |
| ------- | -------------------------------- |
| `Light` | Light grey background, dark icon |
| `Dark`  | Dark blue background, white icon |

---

#### `size: SystemIconCircleSize | number`

> The diameter of the avatar in pixels

Default: `Avatar.Size.m`

| Name  | Size (px) |
| ----- | --------- |
| `xs`  | 16        |
| `s`   | 24        |
| `m`   | 32        |
| `l`   | 40        |
| `xl`  | 64        |
| `xxl` | 120       |

---

#### `altText: string`

> Text describing what the avatar is showing.

---

#### `url: string`

> The URL of the user's photo. Expects a square (1:1) photo.

# Canvas Kit AvatarButton

Button containing an Avatar.

## Usage

```tsx
import * as React from 'react';
import {AvatarButton, AvatarVariant} from '@workday/canvas-kit-react-avatar';

// Basic
<AvatarButton />

// Using static properties on AvatarButton
<AvatarButton
  size={AvatarButton.Size.xs}
  variant={AvatarButton.Variant.Dark}
  onClick={() => { window.alert('AvatarButton Clicked') }}
/>

// Using AvatarVariant import directly
<AvatarButton size={AvatarButton.Size.xs} variant={AvatarVariant.Dark} />
```

## Static Properties

#### `Variant: AvatarVariant`

```tsx
<AvatarButton variant={AvatarButton.Variant.Dark} />
```

---

#### `Size: SystemIconCircleSize | number`

```tsx
<AvatarButton size={AvatarButton.Size.xs} />

// Custom Size.
// Note: Please use the predefined sizes as it works nicely with the Canvas Design System
<AvatarButton size={48} />
```

## Component Props

### Required

> None

### Optional

#### `Variant: AvatarVariant`

> The variant of the avatar if using a default image.

Default: `AvatarVariant.Light`

| Variant | Description                      |
| ------- | -------------------------------- |
| `Light` | Light grey background, dark icon |
| `Dark`  | Dark blue background, white icon |

---

#### `size: SystemIconCircleSize | number`

> The diameter of the avatar in pixels

Default: `AvatarButton.Size.m`

| Name  | Size (px) |
| ----- | --------- |
| `xs`  | 16        |
| `s`   | 24        |
| `m`   | 32        |
| `l`   | 40        |
| `xl`  | 64        |
| `xxl` | 120       |

---

#### `altText: string`

> Text describing what the avatar is showing.

---

#### `url: string`

> The URL of the user's photo. Expects a square (1:1) photo.

---

#### `buttonRef: React.Ref<HTMLButtonElement>`

> A ref to the underlying `<button>` element.

---

#### `onClick: (React.SyntheticEvent) => void`

> A `click` event handler for this component.
