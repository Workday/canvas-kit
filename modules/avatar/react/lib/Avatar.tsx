import * as React from 'react';
import styled from 'react-emotion';
import {colors} from '@workday/canvas-kit-react-core';
import {focusRing, hideMouseFocus} from '@workday/canvas-kit-react-common';
import {SystemIconCircle, SystemIconCircleSize} from '@workday/canvas-kit-react-icon';
import {userIcon} from '@workday/canvas-system-icons-web';

export enum AvatarTheme {
  Light,
  Dark,
}

export interface AvatarProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  /**
   * An AvatarTheme enum indicating which theme to use for the default state (Light vs. Dark)
   */
  themeColor: AvatarTheme;
  /**
   * An AvatarSize enum or number value indicating the size of the avatar
   */
  size: SystemIconCircleSize | number;
  /**
   * Text describing what the avatar is showing
   */
  altText: string;
  /**
   * The url of the users avatar photo
   */
  url?: string;
  /**
   * An event handler function that gets called when the avatar is clicked
   */
  onClick?: (e: React.SyntheticEvent) => void;
  /**
   * Ref of button that the styled component renders.
   */
  buttonRef?: React.Ref<HTMLButtonElement>;
}

const Container = styled('button')<Omit<AvatarProps, 'altText'>>(
  {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 0,
    border: 0,
    borderRadius: '100%',
    boxSizing: 'border-box',
    overflow: 'hidden',
    '& img': {
      width: '100%',
      height: '100%',
    },
  },
  ({themeColor, size, onClick}) => ({
    background: themeColor === AvatarTheme.Dark ? colors.blueberry400 : colors.soap300,
    height: size,
    width: size,
    cursor: onClick ? 'pointer' : 'default',
    '&:not([disabled])': {
      '&:focus': {
        outline: 'none',
        ...(themeColor === AvatarTheme.Dark ? focusRing(2, 2) : focusRing(2)),
      },
    },
    ...hideMouseFocus,
  })
);

export default class Avatar extends React.Component<AvatarProps> {
  static ThemeColor = AvatarTheme;
  static Size = SystemIconCircleSize;

  static defaultProps = {
    themeColor: AvatarTheme.Light,
    size: SystemIconCircleSize.m,
    altText: 'Avatar',
  };

  render() {
    const {buttonRef, themeColor, altText, size, url, onClick, ...elemProps} = this.props;

    const background = themeColor === AvatarTheme.Dark ? colors.blueberry400 : colors.soap300;
    return (
      <Container
        {...elemProps}
        themeColor={themeColor}
        size={size}
        onClick={onClick}
        disabled={onClick ? false : true}
        innerRef={buttonRef}
        aria-label={altText}
      >
        {url ? (
          <img src={url} alt={altText} />
        ) : (
          <SystemIconCircle icon={userIcon} background={background} size={size} />
        )}
      </Container>
    );
  }
}
