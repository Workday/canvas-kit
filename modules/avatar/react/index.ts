import Avatar from './lib/Avatar';

export default Avatar;
export {Avatar};
export * from './lib/Avatar';
