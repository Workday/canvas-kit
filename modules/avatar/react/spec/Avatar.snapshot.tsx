import * as React from 'react';
import Avatar from '../lib/Avatar';
import * as renderer from 'react-test-renderer';

describe('Avatar Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Avatar>Avatar</Avatar>);
    expect(component).toMatchSnapshot();
  });
  test('renders a large Avatar', () => {
    const component = renderer.create(<Avatar size={Avatar.Size.xl} />);
    expect(component).toMatchSnapshot();
  });
  test('renders a large, dark Avatar', () => {
    const component = renderer.create(
      <Avatar size={Avatar.Size.xl} variant={Avatar.Variant.Dark} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders an avatar with a photo', () => {
    const component = renderer.create(
      <Avatar url="https://s3-us-west-2.amazonaws.com/design-assets-internal/avatars/lmcneil.png" />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders a large avatar with a photo', () => {
    const component = renderer.create(
      <Avatar
        size={Avatar.Size.xl}
        url="https://s3-us-west-2.amazonaws.com/design-assets-internal/avatars/lmcneil.png"
      />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders an avatar with an aria label and an alt text', () => {
    const component = renderer.create(
      <Avatar
        altText="my image"
        aria-label="my avatar label"
        size={Avatar.Size.xl}
        url="https://s3-us-west-2.amazonaws.com/design-assets-internal/avatars/lmcneil.png"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
