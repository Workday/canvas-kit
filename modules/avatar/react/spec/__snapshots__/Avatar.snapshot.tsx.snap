// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshots renders a large Avatar 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  height: 64px;
  width: 64px;
  cursor: default;
}

.emotion-2 img {
  width: 100%;
  height: 100%;
}

.emotion-2:not([disabled]):focus {
  outline: none;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='touch'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  width: 64px;
  height: 64px;
}

.emotion-1 img {
  width: 100%;
  height: 100%;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: rgba(0,0,0,0.65);
}

.emotion-0:hover .wd-icon-fill {
  fill: rgba(0,0,0,0.65);
}

.emotion-0 .wd-icon-accent {
  fill: rgba(0,0,0,0.65);
}

.emotion-0:hover .wd-icon-accent {
  fill: rgba(0,0,0,0.65);
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-0 svg {
  width: 40px;
  height: 40px;
}

<button
  aria-label="Avatar"
  className="emotion-2 emotion-3"
  disabled={true}
  size={64}
>
  <div
    className="emotion-1"
    size={64}
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-user wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M20 20.491a.503.503 0 0 1-.496.505H4.496A.493.493 0 0 1 4 20.49v-1.495C4 16 6.246 13.8 8.756 12.683 7.682 11.675 7 10.174 7 8.5 7 5.462 9.239 3 12 3s5 2.462 5 5.499c0 1.675-.68 3.175-1.755 4.183C17.758 13.797 20 15.995 20 18.996v1.495zM6.083 18.996H18c0-2.996-2.5-4.999-6-4.999s-6 2.003-5.917 5zM12 11.998c1.657 0 3-1.567 3-3.5C15 6.567 13.657 5 12 5S9 6.566 9 8.499c0 1.932 1.343 3.499 3 3.499z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </div>
</button>
`;

exports[`Avatar Snapshots renders a large avatar with a photo 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  height: 64px;
  width: 64px;
  cursor: default;
}

.emotion-0 img {
  width: 100%;
  height: 100%;
}

.emotion-0:not([disabled]):focus {
  outline: none;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

[data-whatinput='mouse'] .emotion-0:focus,
[data-whatinput='touch'] .emotion-0:focus,
[data-whatinput='pointer'] .emotion-0:focus {
  outline: none;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

<button
  aria-label="Avatar"
  className="emotion-0 emotion-1"
  disabled={true}
  size={64}
>
  <img
    alt="Avatar"
    src="https://s3-us-west-2.amazonaws.com/design-assets-internal/avatars/lmcneil.png"
  />
</button>
`;

exports[`Avatar Snapshots renders a large, dark Avatar 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #0875e1;
  height: 64px;
  width: 64px;
  cursor: default;
}

.emotion-2 img {
  width: 100%;
  height: 100%;
}

.emotion-2:not([disabled]):focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='touch'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #0875e1;
  width: 64px;
  height: 64px;
}

.emotion-1 img {
  width: 100%;
  height: 100%;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-fill {
  fill: #ffffff;
}

.emotion-0 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0:hover .wd-icon-accent {
  fill: #ffffff;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-0 svg {
  width: 40px;
  height: 40px;
}

<button
  aria-label="Avatar"
  className="emotion-2 emotion-3"
  disabled={true}
  size={64}
>
  <div
    className="emotion-1"
    size={64}
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-user wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M20 20.491a.503.503 0 0 1-.496.505H4.496A.493.493 0 0 1 4 20.49v-1.495C4 16 6.246 13.8 8.756 12.683 7.682 11.675 7 10.174 7 8.5 7 5.462 9.239 3 12 3s5 2.462 5 5.499c0 1.675-.68 3.175-1.755 4.183C17.758 13.797 20 15.995 20 18.996v1.495zM6.083 18.996H18c0-2.996-2.5-4.999-6-4.999s-6 2.003-5.917 5zM12 11.998c1.657 0 3-1.567 3-3.5C15 6.567 13.657 5 12 5S9 6.566 9 8.499c0 1.932 1.343 3.499 3 3.499z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </div>
</button>
`;

exports[`Avatar Snapshots renders an avatar with a photo 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  height: 32px;
  width: 32px;
  cursor: default;
}

.emotion-0 img {
  width: 100%;
  height: 100%;
}

.emotion-0:not([disabled]):focus {
  outline: none;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

[data-whatinput='mouse'] .emotion-0:focus,
[data-whatinput='touch'] .emotion-0:focus,
[data-whatinput='pointer'] .emotion-0:focus {
  outline: none;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

<button
  aria-label="Avatar"
  className="emotion-0 emotion-1"
  disabled={true}
  size={32}
>
  <img
    alt="Avatar"
    src="https://s3-us-west-2.amazonaws.com/design-assets-internal/avatars/lmcneil.png"
  />
</button>
`;

exports[`Avatar Snapshots renders as expected 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  height: 32px;
  width: 32px;
  cursor: default;
}

.emotion-2 img {
  width: 100%;
  height: 100%;
}

.emotion-2:not([disabled]):focus {
  outline: none;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

[data-whatinput='mouse'] .emotion-2:focus,
[data-whatinput='touch'] .emotion-2:focus,
[data-whatinput='pointer'] .emotion-2:focus {
  outline: none;
  box-shadow: none;
  -webkit-animation: none;
  animation: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  border: 0;
  border-radius: 100%;
  box-sizing: border-box;
  overflow: hidden;
  background: #e8ebed;
  width: 32px;
  height: 32px;
}

.emotion-1 img {
  width: 100%;
  height: 100%;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: rgba(0,0,0,0.65);
}

.emotion-0:hover .wd-icon-fill {
  fill: rgba(0,0,0,0.65);
}

.emotion-0 .wd-icon-accent {
  fill: rgba(0,0,0,0.65);
}

.emotion-0:hover .wd-icon-accent {
  fill: rgba(0,0,0,0.65);
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-0 svg {
  width: 20px;
  height: 20px;
}

<button
  aria-label="Avatar"
  className="emotion-2 emotion-3"
  disabled={true}
  size={32}
>
  <div
    className="emotion-1"
    size={32}
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-user wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M20 20.491a.503.503 0 0 1-.496.505H4.496A.493.493 0 0 1 4 20.49v-1.495C4 16 6.246 13.8 8.756 12.683 7.682 11.675 7 10.174 7 8.5 7 5.462 9.239 3 12 3s5 2.462 5 5.499c0 1.675-.68 3.175-1.755 4.183C17.758 13.797 20 15.995 20 18.996v1.495zM6.083 18.996H18c0-2.996-2.5-4.999-6-4.999s-6 2.003-5.917 5zM12 11.998c1.657 0 3-1.567 3-3.5C15 6.567 13.657 5 12 5S9 6.566 9 8.499c0 1.932 1.343 3.499 3 3.499z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </div>
</button>
`;
